src/model/misc.rs

```diff
@@ -19,73 +18,59 @@ pub trait Mentionable {
 
 impl Mentionable for ChannelId {
     fn mention(&self) -> String {
-        format!("{}", self)
+        format!("<#{}>", self.0)
     }
 }
 
 impl Mentionable for Channel {
     fn mention(&self) -> String {
-        format!("{}", self)
+        match self {
```

Comment:
If you deref here (`match *self`), the references (`&`) on each arm can be omitted below